From d372a29bf3fbad783c65e502a40ab19d2ad163ad Mon Sep 17 00:00:00 2001
From: Lu Jialin <lujialin4@huawei.com>
Date: Wed, 19 Oct 2022 10:59:53 +0800
Subject: [PATCH 029/595] [Huawei] cgroup: Fix wb_blkio_write when invalid
 input parameter

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187773

-------------------------------

cgroup1_get_from_id() will not return NULL when input id is not an ino
of a cgroup in cgroupv1. It will return a wrong pointer. Therefore,
modify the judgment condition of cgroup1_get_from_id in wb_blkio_write.

Fixes: 13b8199f0a47 ("cgroup: support cgroup writeback on cgroupv1")
Signed-off-by: Lu Jialin <lujialin4@huawei.com>
---
 mm/memcontrol.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/memcontrol.c b/mm/memcontrol.c
index 88e1c0dc78f2..0a15761ba496 100644
--- a/mm/memcontrol.c
+++ b/mm/memcontrol.c
@@ -5248,7 +5248,7 @@ static ssize_t wb_blkio_write(struct kernfs_open_file *of, char *buf,
 	mutex_lock(&cgroup_mutex);
 	root = blkcg_root_css->cgroup->root;
 	blk_cgroup = cgroup1_get_from_id(root, cgrp_id);
-	if (!blk_cgroup) {
+	if (IS_ERR(blk_cgroup)) {
 		mutex_unlock(&cgroup_mutex);
 		return -EINVAL;
 	}
-- 
2.17.1

